src/DynamoCore/Scheduler/DynamoSchedulerInternals.cs

```diff
@@ -47,6 +47,11 @@ public bool HasPendingTasks
                 }
             }
         }
+
+        /// <summary>
+        /// Returns true if a task is curently executing.
+        /// </summary>
+        public bool HasTaskInProgress { get; set; }
```

Comment:
It doesn't seem these changes are related to this PR, can you remove them?

Reply:
Ok, will remove this and address it in another PR.